### PR TITLE
NET-4993 JWT auth basic acceptance test

### PR DESCRIPTION
**Pull request by jm96441n**
Changes proposed in this PR:
- Adding skipped acceptance test for APIGW JWT work, tests are skipped because they won't pass until JWT work is complete
-

How I've tested this PR:
Ran the tests

How I expect reviewers to test this PR:
Read the code (tests don't currently pass)

Checklist:
- [ ] Tests added
- [ ] [CHANGELOG entry added](https://github.com/hashicorp/consul-k8s/blob/main/CONTRIBUTING.md#adding-a-changelog-entry)

### CODE REVIEW

File: acceptance/tests/api-gateway/api_gateway_test.go
Comment (nathancoleman) [inline acceptance/tests/api-gateway/api_gateway_test.go:1]: Do we need to modify this test to account for JWT on the Gateway being an enterprise-only feature?

File: acceptance/tests/api-gateway/api_gateway_test.go [hunk 1]
```diff
@@ -283,6 +283,251 @@ func TestAPIGateway_Basic(t *testing.T) {
...
+	k8s.DeployKustomize(t, ctx.KubectlOptions(t), cfg.NoCleanupOnFailure, cfg.NoCleanup, cfg.DebugDirectory, "../fixtures/bases/static-client")
+
+	k8s.RunKubectl(t, ctx.KubectlOptions(t), "wait", "--for=condition=available", "--timeout=5m", fmt.Sprintf("deploy/%s", "static-server"))
+	// Grab a kubernetes client so that we can verify binding
+	// behavior prior to issuing requests through the gateway.
+	k8sClient := ctx.ControllerRuntimeClient(t)
+
+	// On startup, the controller can take upwards of 1m to perform
+	// leader election so we may need to wait a long time for
+	// the reconcile loop to run (hence the 1m timeout here).
```
Comment (missylbytes) [inline acceptance/tests/api-gateway/api_gateway_test.go:350]: Nitpick, but this says 1m timeout. Wouldn't it be 2 minutes or do I not understand the retry function

File: acceptance/tests/api-gateway/api_gateway_test.go
Comment (jm96441n) [reply acceptance/tests/api-gateway/api_gateway_test.go:1]: we most definitely do!

File: acceptance/tests/api-gateway/api_gateway_test.go (hunk 1 above)
Comment (jm96441n) [reply acceptance/tests/api-gateway/api_gateway_test.go:350]: you are most definitely right, lemme update this and the spot I copy/pasted from

File: acceptance/tests/api-gateway/api_gateway_test.go [hunk 2]
```diff
@@ -126,7 +126,7 @@ func TestAPIGateway_Basic(t *testing.T) {
 
 			// On startup, the controller can take upwards of 1m to perform
 			// leader election so we may need to wait a long time for
-			// the reconcile loop to run (hence the 1m timeout here).
+			// the reconcile loop to run (hence the 2m timeout here).
```
Comment (nathancoleman) [inline acceptance/tests/api-gateway/api_gateway_test.go:129]: I'd be favor of dropping the implementation-specific bit and avoiding things falling out of sync
```suggestion
			// the reconcile loop to run (hence the timeout here).
```

File: acceptance/tests/api-gateway/api_gateway_test.go [hunk 3]
```diff
@@ -283,6 +283,257 @@ func TestAPIGateway_Basic(t *testing.T) {
...
+	helpers.Cleanup(t, cfg.NoCleanupOnFailure, cfg.NoCleanup, func() {
+		// Ignore errors here because if the test ran as expected
+		// the custom resources will have been deleted.
+		k8s.RunKubectlAndGetOutputE(t, ctx.KubectlOptions(t), "delete", "-k", "../fixtures/cases/api-gateways/jwt-auth")
+	})
+
+	// Create certificate secret, we do this separately since
+	// applying the secret will make an invalid certificate that breaks other tests
+	logger.Log(t, "creating certificate secret")
+	out, err = k8s.RunKubectlAndGetOutputE(t, ctx.KubectlOptions(t), "apply", "-f", "../fixtures/bases/api-gateway/certificate.yaml")
```
Comment (nathancoleman) [inline acceptance/tests/api-gateway/api_gateway_test.go:331]: `.../bases/...` is soooo close to `.../cases/...` that I was almost certain I'd found a typo 🤣

File: acceptance/tests/api-gateway/api_gateway_test.go [hunk 4]
```diff
@@ -283,6 +283,257 @@ func TestAPIGateway_Basic(t *testing.T) {
...
+	k8sClient := ctx.ControllerRuntimeClient(t)
+
+	// On startup, the controller can take upwards of 1m to perform
+	// leader election so we may need to wait a long time for
+	// the reconcile loop to run (hence the 2m timeout here).
+	var (
+		gatewayAddress string
+		gatewayClass   gwv1beta1.GatewayClass
+		httproute      gwv1beta1.HTTPRoute
+		httprouteAuth  gwv1beta1.HTTPRoute
```
Comment (nathancoleman) [inline acceptance/tests/api-gateway/api_gateway_test.go:361]: #nit
```suggestion
		httpRoute      gwv1beta1.HTTPRoute
		httpRouteAuth  gwv1beta1.HTTPRoute
```

File: acceptance/tests/fixtures/cases/api-gateways/jwt-auth/httproute.yaml
Comment (nathancoleman) [inline acceptance/tests/fixtures/cases/api-gateways/jwt-auth/httproute.yaml:1]: I would name this file `httproute-auth.yaml` to correspond with the variable name in the test

File: acceptance/tests/fixtures/cases/api-gateways/jwt-auth/httproute-no-auth.yaml
Comment (nathancoleman) [inline acceptance/tests/fixtures/cases/api-gateways/jwt-auth/httproute-no-auth.yaml:1]: I would name this file `httproute.yaml` to correspond with the variable name in the test

File: acceptance/tests/fixtures/cases/api-gateways/jwt-auth/httproute.yaml [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+          type: PathPrefix
+          value: "/admin"
+      backendRefs:
+        - name: static-server
+          port: 80
+      filters:
+      - type: ExtensionRef
+        extensionRef:
+          group: consul.hashicorp.com
+          kind: HTTPAuthFilter
```
Comment (nathancoleman) [inline acceptance/tests/fixtures/cases/api-gateways/jwt-auth/httproute.yaml:24]: Should this be `HTTPRouteAuthFilter` based on #2734 ?

File: acceptance/tests/fixtures/cases/api-gateways/jwt-auth/jwt-route-filter.yaml [hunk 6]
```diff
@@ -0,0 +1,12 @@
+apiVersion: consul.hashicorp.com/v1alpha1
+kind: HTTPAuthFilter
```
Comment (nathancoleman) [inline acceptance/tests/fixtures/cases/api-gateways/jwt-auth/jwt-route-filter.yaml:2]: Should this be `HTTPRouteAuthFilter` based on #2734 ?

File: acceptance/tests/api-gateway/api_gateway_test.go [hunk 7]
```diff
@@ -283,6 +283,257 @@ func TestAPIGateway_Basic(t *testing.T) {
 	}
 }
 
+func TestAPIGateway_JWTAuth_Basic(t *testing.T) {
+	t.Skip()
```
Comment (nathancoleman) [inline acceptance/tests/api-gateway/api_gateway_test.go:287]: Would add a note about why we're skipping for now and will enable once implementation is complete

**Review by nathancoleman — APPROVED** [review]
Love seeing this test come to life so early on! Nice work 🏆

File: acceptance/tests/fixtures/cases/api-gateways/jwt-auth/jwt-route-filter.yaml (hunk 6 above)
Comment (jm96441n) [reply acceptance/tests/fixtures/cases/api-gateways/jwt-auth/jwt-route-filter.yaml:2]: yup it should be!

**Pull request merged**